### PR TITLE
Dev 2(string analyzer)

### PR DESCRIPTION
**Pull request by PavelBadeha**

### CODE REVIEW

File: Dev-2/Dev-2/StringAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+{
+    public class StringAnalyzer
+    {
+
+        /// <summary>
+        /// Returns the maximum number of unequal consecutive characters in a string
+        /// </summary>
+        public int MaxOfNotIdenticalConsecutiveSymbols(string str)
+        {
+            int result = 1;
```
Comment (IgorKheidorov) [inline Dev-2/Dev-2/StringAnalyzer.cs:15]: Why buff? better maxCount

File: Dev-2/Dev-2/StringAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
+﻿using System.Linq;
+
+namespace Dev_2
+{
+    public class StringAnalyzer
+    {
+
+        /// <summary>
+        /// Returns the maximum number of unequal consecutive characters in a string
```
Comment (IgorKheidorov) [inline Dev-2/Dev-2/StringAnalyzer.cs]: No check for null string

File: Dev-2/Dev-2/StringAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            if (str == string.Empty)
+            {
+                return 0;
+            }
+            return result;                   
+        }
+
+        /// <summary>
+        /// Returns the maximum number of identical consecutive digits in a string
+        /// </summary>
```
Comment (IgorKheidorov) [inline Dev-2/Dev-2/StringAnalyzer.cs:45]: Much copy paste in three almost identical methods, try to add one common method for search and use it inside MaxOfIdenticalConsecutiveDigits and others

File: Dev-2/Dev-2UnitTests/TesterOfStringAnalyzer.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Dev_2UnitTests
+{
+    /// <summary>
+    /// TestClass for StringAnalyzer
+    /// </summary>
```
Comment (IgorKheidorov) [inline Dev-2/Dev-2UnitTests/TesterOfStringAnalyzer.cs:7]: Add "Negative" or "Positive" in test names? check for null string

**Review by IgorKheidorov — APPROVED** [review]
